examples/blocked_cholesky_multigpu.py

```diff
@@ -0,0 +1,265 @@
+          set_gpu_memory_from_gpu(i, j, num_gpus, out)
+
+      # Triangular solve
+      @spawn(solve[i, j], [gemm2[i, j, 0:j], subcholesky[j]], placement=[gpu(i%num_gpus)])
+      def t4():
+        factor = get_gpu_memory(j, j, num_gpus)
+        panel  = get_gpu_memory(i, j, num_gpus)
+        out = ltriang_solve(factor, panel)
+        set_gpu_memory_from_gpu(i, j, num_gpus, out)
+
+  return subcholesky[a.shape[0]-1]
+
+def allocate_gpu_memory(i:int, r:int, n:int, b:int):
+  with cp.cuda.Device(i):
+    logger.debug("\tAllocate device:", i, "...")
+    prealloced = cp.ndarray([r, n // b, b, b])
+    gpu_arrs.append(prealloced)
+
+def get_gpu_memory(i:int, j:int, num_gpus:int):
+  dev_id   = i % num_gpus
+  local_id = i // num_gpus
+  src = gpu_arrs[dev_id][local_id][j]
+  dst = clone_here(src)
+  return dst
+
+def set_gpu_memory_from_gpu(i:int, j:int, num_gpus:int, v):
+  dev_id   = i % num_gpus
+  local_id = i // num_gpus
+  gpu_arrs[dev_id][local_id][j] = v
+
+def set_gpu_memory_from_cpu(a, num_gpus):
+  for j in range(a.shape[0]):
+    dev_id   = j % num_gpus 
+    local_id = j // num_gpus 
+    with cp.cuda.Device(dev_id):
+      gpu_arrs[dev_id][local_id] = cp.array(a[j], copy=True)
+
+def main():
+  num_gpus = cp.cuda.runtime.getDeviceCount()
+  @spawn(placement=cpu)
+  async def test_blocked_cholesky():
+    logger.debug("Block size=", block_size, " and total array size=", n)
+    assert not n % block_size
+
+    logger.debug("Allocate memory..")
+    for d in range(num_gpus):
+      row_size = n // (block_size * num_gpus)
+      if d < ((n / block_size) % num_gpus):
+        row_size += 1
+      if row_size > 0:
+        allocate_gpu_memory(d, row_size, n, block_size)
+    logger.debug("Allocate memory done..")
+
+    np.random.seed(10)
+
+    logger.debug("Random number generate..")
+    # Construct input data
+    a = np.random.rand(n, n)
+    a = a @ a.T
+    logger.debug("Random number generate done..")
+
+    logger.debug("Copy a to a1..")
+    # Copy and layout input
+    a1 = a.copy()
+    logger.debug("Copying done..")
+    logger.debug("Shaping starts..")
+    ap = a1.reshape(n // block_size, block_size, n // block_size, block_size).swapaxes(1,2)
+    logger.debug("Shaping done..")
+    set_gpu_memory_from_cpu(ap, num_gpus)
+
+    for i in range(len(gpu_arrs)):
+      logger.debug("Device ", i, " arrays are on ", gpu_arrs[i].device);
+    logger.debug("Calculate starts..")
+    start = time.perf_counter()
+
+    # Call Parla Cholesky result and wait for completion
+    await cholesky_blocked_inplace(ap, num_gpus)
+
+    end = time.perf_counter()
+    print(end - start, "seconds")
+    logger.debug("Calculate done..")
+
+    for i in range(len(gpu_arrs)):
+      logger.debug("Device ", i, " arrays are on ", gpu_arrs[i].device);
+
+    for d in range(len(gpu_arrs)):
+      logger.debug("Device:", d, " swap array..")
+      gpu_arrs[d] = cp.swapaxes(gpu_arrs[d], 2, 1)
+      logger.debug("Device:", d, " swap done..")
```

Comment:
Why not use `@spawn` and get some parallelism here. :-)

Reply:
It looks very good point. But for now, if you don't mind, let me just postpone this work to the another PR because it is excluded from time measurement and some people are waiting for this PR.